test/logflare_web/live/billingaccount_live/estimate_usage_component_test.exs

```diff
@@ -0,0 +1,51 @@
+defmodule LogflareWeb.BillingAccountLive.EstimateUsageComponentTest do
+  use Logflare.DataCase
+  alias LogflareWeb.BillingAccountLive.EstimateUsageComponent
+  import Phoenix.LiveViewTest
+  @endpoint LogflareWeb.Endpoint
+
+  describe "EstimateUsageComponent" do
+    setup do
+      user = insert(:user, billing_enabled: true)
+      plan = insert(:plan, price: TestUtils.random_pos_integer())
+      end_date = DateTime.utc_now()
+
+      start_date =
+        end_date
+        |> then(&Date.new!(&1.year, &1.month, &1.day))
+        |> Date.beginning_of_month()
+        |> DateTime.new!(~T[00:00:00])
+
+      billing_counts = insert_list(2, :billing_counts, user: user, inserted_at: start_date)
+
+      %{
+        user: user,
+        plan: plan,
+        billing_counts: billing_counts,
+        start_date: start_date,
+        end_date: end_date
+      }
+    end
+
+    test "renders element with estimated usage and price for a given period and plan", %{
+      user: user,
+      plan: plan,
+      billing_counts: billing_counts,
+      start_date: start_date,
+      end_date: end_date
+    } do
+      expected_presentation_start_date = Calendar.strftime(start_date, "%b %d")
+      expected_presentation_end_date = Calendar.strftime(end_date, "%b %d")
+      expected_usage = billing_counts |> Enum.map(& &1.count) |> Enum.sum()
+      expected_cost = expected_usage * plan.price
+
+      result = render_component(EstimateUsageComponent, %{user: user, plan: plan})
+
+      assert result =~
+               "Estimate usage and cost between #{expected_presentation_start_date} and #{expected_presentation_end_date}:"
+
+      assert result =~ "Usage: #{expected_usage} inserts"
+      assert result =~ "Estimated Cost: #{expected_cost} USD"
```

Comment:
don't need to test for exact text, can do a Regex match instead. main thing we're testing for is that estimated cost is presented, actual value has already been covered by other tests.
`~r/Estimated cost: [0-9]+ USD/` (and a similar one for inserts) would check that everything works with less setup needed.
What you're doing above is testing the implementation, when we'd want to instead test for end result (i.e. that something is rendered and the rendering doesn't break)

So for example, we can simplify the start/end date assertion to a regex test
```
assert Regex.match?(~r/Estimated usage and cost between .+ and .+/, result)
```
the actual calendar dates are irrelevant and would be testing for implementation.

Reply:
We do have the cumulative usage tested elsewhere but the cost is not yet, that is done by getting the user's plan and multiplying on the view so that it isn't tested at the moment in any other spot except here and probably we should keep it that way for now to avoid more complexity

Reply:
Then we should shift the cost calculation logic down from there view to the context instead, where it'll be easier to test and isolate.

The test logic is quite complex and hard to read in its current state, I'm mostly concerned about the test code maintenance. We should be testing for whether the view layer breaks, and not for biz logic which is already covered in the context. 

As this is a purely presentational liveview component, testing that text gets rendered and doesn't crash out is enough

Reply:
honestly don't see this being used elsewhere but will change to be more inline with code expectations

Reply:
It's fine, the table is a temp measure in any case, we can pick up billing ui improvements post LW

Reply:
ended up doing the change, didn't see this reply 🤦 😅